docs/core/whats-new/dotnet-10/libraries.md

```diff
@@ -0,0 +1,267 @@
+
+Previously, we introduced the following method without adding an overload that takes a single parameter:
+
+```C#
+public static TimeSpan FromMilliseconds(long milliseconds, long microseconds = 0);
+```
+
+Although this works since the second parameter is optional, it causes a compilation error when used in a LINQ expression like:
+
+```C#
+Expression<Action> a = () => TimeSpan.FromMilliseconds(1000);
+```
+
+The issue arises because LINQ expressions cannot handle optional parameters. To address this, we are introducing an overload that takes a single parameter and modifying the existing method to make the second parameter mandatory:
+
+```C#
+public readonly struct TimeSpan
+{
+    public static TimeSpan FromMilliseconds(long milliseconds, long microseconds); // Second parameter is no longer optional
+    public static TimeSpan FromMilliseconds(long milliseconds);  // New overload
+}
+```
+
+## ZipArchive performance and memory improvements
+
+Two significant PRs have been made by contributor @edwardneal in .NET 10 Preview 1 to improve the performance and memory usage of `ZipArchive`:
+
+- [dotnet/runtime #102704](https://github.com/dotnet/runtime/pull/102704) optimizes the way entries are written to a `ZipArchive` when in `Update` mode. Previously, all `ZipArchiveEntries` would be loaded into memory and rewritten, which could lead to high memory usage and performance bottlenecks. The optimization reduces memory usage and improves performance by avoiding the need to load all entries into memory.
+
+Adding a 2GB zip file to an existing archive showed:
+
+- A 99.8% reduction in execution time.
+- A 99.9996% reduction in memory usage.
+
+### Benchmarks:
+
+| Method    | Job      | Runtime   | Mean     | Error      | StdDev      | Ratio | RatioSD | Gen0      | Gen1      | Gen2      | Allocated | Alloc Ratio |
+|-----------|----------|-----------|---------:|-----------:|------------:|------:|--------:|----------:|----------:|----------:|----------:|------------:|
+| Benchmark | Baseline | .NET 9.0  | 4.187 s  | 83.3751 ms | 177.6792 ms | 1.002 | 0.06    | 1000.0000 | 1000.0000 | 1000.0000 | 2 GB      | 1.000       |
+| Benchmark | CoreRun  | .NET 10.0 | 9.452 ms | 0.1583 ms  | 0.1322 ms   | 0.002 | 0.00    | -         | -         | -         | 7.01 KB   | 0.000       |
+
+Additional details are provided in [dotnet/runtime #102704](https://github.com/dotnet/runtime/pull/102704#issue-2317941700).
+
+- [dotnet/runtime #103153](https://github.com/dotnet/runtime/pull/103153) enhances the performance of `ZipArchive` by parallelizing the extraction of entries and optimizing internal data structures for better memory usage. These improvements address issues related to performance bottlenecks and high memory usage, making `ZipArchive` more efficient and faster, especially when dealing with large archives.
+
+Reading a zip archive showed:
+
+- An 18% reduction in execution time.
+- An 18% reduction in memory usage.
+
+### Benchmarks:
```

Comment:
This is a lot more detailed info than I usually put in the what's new docs. I would suggest removing this section and just saying something like "For more information and detailed benchmarks, see ...".